auth.js

```diff
@@ -1,4 +1,14 @@
-const request = require('request-promise');
+const { requestWithRetries, isRequestError, isStatusCodeError } = require('./requestPromiseUtils');
+
+const retryHandler = (err, nextTryCount) => {
```

Comment:
Wondering if we could name this more clearly, like `shouldRetry` to reflect more of a bool return.

To me, a `handler` implies it will actually do the handling.

Reply:
`retryComputor` was an early candidate ... :)

Reply:
`retryDecider` was also pretty high on the list